restapi/src/main/java/io/stargate/web/docsapi/service/DocsSchemaChecker.java

```diff
@@ -0,0 +1,53 @@
+package io.stargate.web.docsapi.service;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import io.stargate.db.schema.Schema;
+import io.stargate.web.docsapi.dao.DocumentDB;
+import io.stargate.web.docsapi.exception.ErrorCode;
+import io.stargate.web.docsapi.exception.ErrorCodeRuntimeException;
+import io.stargate.web.docsapi.service.util.ImmutableKeyspaceAndTable;
+
+/**
+ * A helper class for checking that particular keyspace/table combinations exist and are valid
+ * document API collections.
+ */
+public class DocsSchemaChecker {
+  private Schema lastCheckedSchema;
+
+  private final Cache<ImmutableKeyspaceAndTable, Boolean> validatedDocCollections =
+      Caffeine.newBuilder().maximumSize(10_000).build();
```

Comment:
nit: For this simple use case a `Caffeine` cache might be an overkill. A simple `ConcurrentHashMap` could do. WDYT?

Reply:
Oh yeah that's true - since there are no time-based invalidation rules or anything